subprojects/testing-native/src/main/java/org/gradle/nativeplatform/test/xctest/plugins/XCTestConventionPlugin.java

```diff
@@ -0,0 +1,164 @@
+        project.getPluginManager().apply(SwiftBasePlugin.class);
+
+        final DirectoryVar buildDirectory = project.getLayout().getBuildDirectory();
+        Directory projectDirectory = project.getLayout().getProjectDirectory();
+        ConfigurationContainer configurations = project.getConfigurations();
+        TaskContainer tasks = project.getTasks();
+
+        // Add the component extension
+        SwiftComponent component = project.getExtensions().create(SwiftComponent.class, "xctest", DefaultSwiftComponent.class, fileOperations);
+        component.getSource().from(projectDirectory.dir("src/test/swift"));
+
+        // Add a compile task
+        SwiftCompile compile = tasks.create("compileTestSwift", SwiftCompile.class);
+
+        compile.includes(configurations.getByName(SwiftBasePlugin.SWIFT_TEST_IMPORT_PATH));
+
+        FileCollection sourceFiles = component.getSwiftSource();
+        compile.source(sourceFiles);
+
+        File frameworkDir = new File(getSdkPlatformPath(), "Developer/Library/Frameworks");
+
+        compile.setCompilerArgs(Lists.newArrayList("-g", "-F" + frameworkDir.getAbsolutePath()));
+        compile.setMacros(Collections.<String, String>emptyMap());
+        compile.setModuleName(project.getName());
+
+        compile.setObjectFileDir(buildDirectory.dir("test/objs"));
+
+        DefaultNativePlatform currentPlatform = new DefaultNativePlatform("current");
+        compile.setTargetPlatform(currentPlatform);
+
+        // TODO - make this lazy
+        NativeToolChain toolChain = project.getModelRegistry().realize("toolChains", NativeToolChainRegistryInternal.class).getForPlatform(currentPlatform);
+        compile.setToolChain(toolChain);
+
+        // Add a link task
+        LinkExecutable link = tasks.create("linkTest", LinkExecutable.class);
+        // TODO - need to set basename
+        link.source(compile.getObjectFileDirectory().getAsFileTree().matching(new PatternSet().include("**/*.obj", "**/*.o")));
+        link.lib(configurations.getByName(CppBasePlugin.NATIVE_TEST_LINK));
+        link.setLinkerArgs(Lists.newArrayList("-Xlinker", "-bundle", "-F" + frameworkDir.getAbsolutePath(), "-framework", "XCTest", "-Xlinker", "-rpath", "-Xlinker", "@executable_path/../Frameworks", "-Xlinker", "-rpath", "-Xlinker", "@loader_path/../Frameworks"));
+        PlatformToolProvider toolProvider = ((NativeToolChainInternal) toolChain).select(currentPlatform);
+        Provider<RegularFile> exeLocation = buildDirectory.file(toolProvider.getExecutableName("exe/" + project.getName() + "Test"));
+        link.setOutputFile(exeLocation);
+        link.setTargetPlatform(currentPlatform);
+        link.setToolChain(toolChain);
+
+
+        Provider<Directory> testBundleDir = buildDirectory.dir("bundle/" + project.getName() + "Test.xctest");
+        final CreateXcTestBundle testBundle = tasks.create("createXcTestBundle", CreateXcTestBundle.class);
+        testBundle.dependsOn(link);
```

Comment:
this should already be inferred by attaching the link task's binary file as input to this task.